sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidRexExecutor.java

```diff
@@ -172,7 +172,8 @@ public void reduce(
           // complex constant is not reducible, so just leave it as an expression
           literal = constExp;
         } else {
-          if (exprResult.isArray()) {
+          if (!exprResult.type().isPrimitive()) {
+            // just leave complex expresions alone
             // just leave array expressions on multi-value strings alone, we're going to push them down into a virtual
             // column selector anyway
```

Comment:
It seems like the new comment line should actually be an edit to the existing comment line that explains that we are leaving all complex expressions alone.  With just adding the new line, the whole set of comments doesn't make sense to a reader that doesn't know the history of changes.

Reply:
i'll try to clear it up a bit, complex expressions and array expressions are separate, we just want to ignore them both for slightly different reasons

Reply:
actually this adjustment was strange, immediately above this block there was some stuff to already do the same thing for complex types, but it was a bit too restrictive, so i've just changed it so that complex types are all handled in that block instead